projects/ngx-drag-to-select/src/lib/select-item.directive.ts

```diff
@@ -44,16 +57,16 @@ export class SelectItemDirective implements OnInit, DoCheck {
     return this._boundingClientRect;
   }
 
-  select() {
-    this.selected = true;
+  calculateBoundingClientRect() {
+    this._boundingClientRect = calculateBoundingClientRect(this.host.nativeElement);
   }
 
-  deselect() {
-    this.selected = false;
+  _select() {
```

Comment:
What about `select()` and `deselect()` without the underscore. 
Because _ would mean private and here it's a public field, probably because it's for internal use only?

Reply:
Yes I have added those underscores to indicate that the are used internally and not meant to be used by the user as these API may change. It's only public because it's used by the container but it should only be used by the container. It doesn't make much sense to call these methods on your own because they simply apply the styling, that's it. There is no connection from the directive to the parent, so no tight coupling between child and parent.